crates/bevy_asset/src/lib.rs

```diff
@@ -1765,8 +1766,11 @@ mod tests {
     #[derive(Asset, TypePath)]
     pub struct TestAsset;
 
-    #[allow(dead_code)]
     #[derive(Asset, TypePath)]
+    #[expect(
+        dead_code,
+        reason = "This exists to ensure that `#[derive(Asset)]` works on enums. The inner variants are known not to be used."
+    )]
```

Comment:
I have no idea why this only lints on the enum test case, despite the struct test cases also being unused.